LogicClass.py

```diff
@@ -201,11 +203,44 @@ def getNotified(self, origin):
         The purpose of this depends on the DHT.
         Example in Chord: I exist and I think you're my successor.
         """
-        #print("GOT NOTIFIED",origin)
+        ##print("GOT NOTIFIED",origin)
         with self.notifiedLock:
             self.notifiedMe.append(origin)
         return True
 
+    def onResponsibilityChange(self):
```

Comment:
It might be worth extending it into a "trigger" that can be assigned more programatically for easy extension later. (so people can register more functions to this event without having to re-write/overwrite the current function)